src/GoodsList.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+class GoodsList extends React.Component {
+  state = {
+    goodsList: [...this.props.goodsList],
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      goodsList: state.goodsList.reverse(),
+    }));
+  }
```

Comment:
```suggestion
  reverse = () => {
    this.setState(state => ({
      goodsList: [...state.goodsList].reverse(),
    }));
  }
```
You're still mutating the state. 'reverse` and `sort` are mutating methods. Create a copy and make all operations with it